### PR TITLE
Creating+initializing the data model  for bookmark feature

### LINKED ISSUE #21: Favorite link list-Data model classes
Create the model classes that describe the data model of the application.

### LINKED ISSUE #22: Favorite link list- Migration files for initializing the data model
We need to create the database table that stores the data it represents for this feature. In Django, we do it via “database migrations”. Database migrations are scripts that can create or alter the database structure.

### PR DESCRIPTION
**Pull request by noaramon**
This PR creates and initialize the data model for the bookmark feature in our app.
Once all the classes in models.py will be created, I'll have to update it again.
fix #21
fix #22

### CODE REVIEW

**Comment by Omeramsc** [comment]
Please elaborate in the description of what this PR should achieve,
also - add `fix` before the link to each issue.

Do note that the PR fails the CI due to flake8 - reformat the code so it'll pass it.

**Comment by noaramon** [comment]
Sorry for the multiple commits, it's because my current IDE don't show these kinds of errors.

**Comment by ravidbro** [comment]
Your branch has conflicts that need to be resolved

File: StuDash_home/models.py [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+# course= models.ForeignKey(Course) will be added to this class
+# after class Course will be created by other team member
+# descripition- Users are related to certain courses.
+# for each course, the user is able to create relevant bookmarks
+# each bookmark stores url, name and last visit date in order to enable potential future filtering
 
-# Create your models here.
+
+class Bookmark (models.Model):
+    user = models.ManyToManyField(User, blank=True)
```
Comment (ravidbro) [inline StuDash_home/models.py]: Can the same bookmark belong to several users?
For example, can one user create a bookmark, and a second user will use it?

File: StuDash_home/models.py [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+    url = models.URLField(max_length=200, primary_key=True)
```
Comment (ravidbro) [inline StuDash_home/models.py]: Why do you want that field to be a primary key?
The table usually has an ID column (I think default by Django generates it), which is the primary key.

**Comment by noaramon** [comment]
As for the first question-yes,  two users can have the same url saved in
their bookmark.

As for the second question- I wasn’t sure in which cases we’ll let Django
to generate it’s own serial id and in which cases we need to specify
primary key.
If the Boomark’s primary key will be an id, i’ll have to change the
relation between user and bookmark to one to many?

**Comment by ravidbro** [comment]
I don't think that if the primary key is the implicit ID column, it somehow affects your relation. It can still be many-to-many if you think it's the right thing.
Looking at Sim's presentation, we can see that he mentions only the class name and not a field in it when defining the foreign key.

About the use here in many-to-many, I'm still not sure I'm convinced.
Some users can have the same URL in their bookmark, but is it really the same URL object in the DB for all of them?
Can you describe the user story or the flow of 2 users? How are they using the system in a way that will lead to that?

**Comment by Omeramsc** [comment]
-```As for the first question-yes,  two users can have the same url saved in
their bookmark.```

That's ok, but as Ravid asked - is it **the same** bookmark or two different bookmarks with the same URL? 
since the bookmarks have a chosen name, and a `last used` property, I think it may be the latter.

**Pull request closed without merging**